src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/MessageBody.cs

```diff
@@ -257,25 +257,7 @@ protected virtual void OnConsumedBytes(int count)
                 }
             }
 
-            if (keepAlive)
-            {
-                return new ForContentLength(true, 0, context);
-            }
-
-            return new ForRemainingData(context);
-        }
-
-        private class ForRemainingData : MessageBody
```

Comment:
Like I mentioned in https://github.com/aspnet/KestrelHttpServer/issues/1104#issuecomment-247186700 We need this for Upgrade requests that usually don't have a `Content-Length` (e.g. WebSockets).